libctru/source/services/am.c

```diff
@@ -89,96 +108,97 @@ Result AM_StartDlpChildCiaInstall(Handle *ciahandle)
 
 	if((ret = svcSendSyncRequest(amHandle))!=0) return ret;
 
-	*ciahandle = cmdbuf[3];
+	*ciaHandle = cmdbuf[3];
 	
 	return (Result)cmdbuf[1];
 }
 
-Result AM_CancelCIAInstall(Handle *ciahandle)
+Result AM_CancelCIAInstall(Handle *ciaHandle)
 {
 	Result ret = 0;
 	u32 *cmdbuf = getThreadCommandBuffer();
 
 	cmdbuf[0] = 0x04040002;
 	cmdbuf[1] = 0x10;
-	cmdbuf[2] = *ciahandle;
+	cmdbuf[2] = *ciaHandle;
 
 	if((ret = svcSendSyncRequest(amHandle))!=0) return ret;
 
 	return (Result)cmdbuf[1];
 }
 
-Result AM_FinishCiaInstall(u8 mediatype, Handle *ciahandle)
+Result AM_FinishCiaInstall(u8 mediatype, Handle *ciaHandle)
 {
 	Result ret = 0;
 	u32 *cmdbuf = getThreadCommandBuffer();
 
 	cmdbuf[0] = 0x04050002;
 	cmdbuf[1] = 0x10;
-	cmdbuf[2] = *ciahandle;
+	cmdbuf[2] = *ciaHandle;
 
 	if((ret = svcSendSyncRequest(amHandle))!=0) return ret;
 
 	return (Result)cmdbuf[1];
 }
 
-Result AM_DeleteTitle(u8 mediatype, u64 titleid)
+Result AM_DeleteTitle(u8 mediatype, u64 titleID)
 {
 	Result ret = 0;
 	u32 *cmdbuf = getThreadCommandBuffer();
 
 	cmdbuf[0] = 0x041000C0;
 	cmdbuf[1] = mediatype;
-	cmdbuf[2] = titleid & 0xffffffff;
-	cmdbuf[3] = (titleid >> 32) & 0xffffffff;
+	cmdbuf[2] = titleID & 0xffffffff;
+	cmdbuf[3] = (u32)(titleID >> 32);
 
 	if((ret = svcSendSyncRequest(amHandle))!=0) return ret;
 
 	return (Result)cmdbuf[1];
 }
 
-Result AM_DeleteAppTitle(u8 mediatype, u64 titleid)
+Result AM_DeleteAppTitle(u8 mediatype, u64 titleID)
 {
 	Result ret = 0;
 	u32 *cmdbuf = getThreadCommandBuffer();
 
 	cmdbuf[0] = 0x000400C0;
 	cmdbuf[1] = mediatype;
-	cmdbuf[2] = titleid & 0xffffffff;
-	cmdbuf[3] = (titleid >> 32) & 0xffffffff;
+	cmdbuf[2] = titleID & 0xffffffff;
+	cmdbuf[3] = (u32)(titleID >> 32);
 
 	if((ret = svcSendSyncRequest(amHandle))!=0) return ret;
 
 	return (Result)cmdbuf[1];
 }
 
-Result AM_InstallFIRM(u8 mediatype, u64 titleid)
+Result AM_InstallFIRM(u64 titleID)
 {
 	Result ret = 0;
 	u32 *cmdbuf = getThreadCommandBuffer();
 
-	cmdbuf[0] = 0x000400C0;
-	cmdbuf[1] = mediatype;
-	cmdbuf[2] = titleid & 0xffffffff;
-	cmdbuf[3] = (titleid >> 32) & 0xffffffff;
+	cmdbuf[0] = 0x04010080;
+	cmdbuf[1] = titleID & 0xffffffff;
+	cmdbuf[2] = (u32)(titleID >> 32);
 
 	if((ret = svcSendSyncRequest(amHandle))!=0) return ret;
 
 	return (Result)cmdbuf[1];
 }
 
-Result AM_GetTitleProductCode(u8 mediatype, u64 titleid, char* productcode)
+Result AM_GetTitleProductCode(u8 mediatype, u64 titleID, char* productCode)
 {
 	Result ret = 0;
 	u32 *cmdbuf = getThreadCommandBuffer();
 	
 	cmdbuf[0] = 0x000500C0;
 	cmdbuf[1] = mediatype;
-	cmdbuf[2] = titleid & 0xffffffff;
-	cmdbuf[3] = (titleid >> 32) & 0xffffffff;
+	cmdbuf[2] = titleID & 0xffffffff;
+	cmdbuf[3] = (u32)(titleID >> 32);
 	
 	if((ret = svcSendSyncRequest(amHandle))!=0) return ret;
 
-	snprintf(productcode, 16, "%s", (char*)(&cmdbuf[2]));
+	// The product code string can use the full 16 bytes without NULL terminator
+	if(productCode) snprintf(productCode, 16, "%s", (char*)&cmdbuf[2]);
```

Comment:
Being honest I think I prefer the memcpy. Is there an advantage to using snprintf here?